content.en/tools/nebula/_index.md

```diff
@@ -0,0 +1,108 @@
+![Nebula Logo](./nebula.png)
+
+# Nebula
+
+Nebula is a [libp2p](https://github.com/libp2p/go-libp2p-kad-dht) DHT crawler
+and monitor that tracks the liveliness of peers. The crawler connects
+to [DHT](https://en.wikipedia.org/wiki/Distributed_hash_table) bootstrap peers
+and then recursively follows all entries in
+their [k-buckets](https://en.wikipedia.org/wiki/Kademlia) until all peers have
+been visited. The crawler supports
+the [IPFS](https://ipfs.network), [Filecoin](https://filecoin.io), [Polkadot](https://polkadot.network/), [Kusama](https://kusama.network/), [Rococo](https://substrate.io/developers/rococo-network/), [Westend](https://wiki.polkadot.network/docs/maintain-networks#westend-test-network)
+networks [and more](https://github.com/dennis-tra/nebula/blob/a33a5fd493caaeb07e92ecc73c32ee87ae9e374f/pkg/config/config.go#L11).
+
+{{< button href="https://github.com/dennis-tra/nebula" >}}GitHub{{< /button >}}
+
+## Components
+
+Nebula is split into two components 1) a **crawler** and 2) a **monitor**. The
+crawler
+is responsible for discovering peers in the DHT and the monitor periodically
+connects to the peers to track their uptime.
+
+## How does it work?
+
+The Kademlia DHT network is a distributed system where each peer maintains a
+routing table containing other peers in the network with specific XOR distances
+to itself. Peers are grouped into "k-buckets" based on the number of shared
+prefix bits between their
+own [`PeerID`](https://docs.libp2p.io/concepts/peer-id/) and the remote PeerID.
+For instance, bucket
+0 includes nodes without any shared prefix bits, and each bucket contains a
+maximum of 20 entries.
+
+To begin the crawl process, Nebula connects to a set of bootstrap peers and
+constructs a routing table for each peer it connects to. Next, it generates
+random keys with common prefix lengths that fall into each peer's buckets and
+asks remote peers if they know any peers that are closer (in XOR distance) to
+the ones Nebula just constructed. This provides a list of all PeerIDs that a
+peer has in its routing table. The process repeats for all found peers until no
+new PeerIDs are found.
+
+If configured to store results in a database, Nebula persists information for
+every visited peer. This information includes latency measurements, current
+multi-addresses, agent version, and supported protocols. If the peer is
+dialable, Nebula also creates a session instance that contains information about
+the node's uptime.
+
+The monitoring process periodically queries the database for these sessions and
+tries to dial the comprised peers. If the peer is dialable it updates the
+session row with the new uptime and if the peer was not dialable it "closes" the
+session. This allows for precise peer churn measurements.
+
```

Comment:
I think we should define the concepts used by Nebula more clearly and not just in terms of the database operation. For example, I wrote this about a session on the dht page:

> When a new peer is discovered, the crawler records the start of a session of availability and extends the session length with every successful connection attempt. However, a failed connection terminates the session, and a later successful attempt starts a new session. Peers can have multiple sessions of availability during each measurement period.

It would be good to expand on that here. 

We have charts that mention connection errors, so we could also add information here about the types of errors that Nebula records when crawling and monitoring. 

Also, perhaps mention that we record peer IDs and IP addresses. We have plots that show differing numbers of ip addresses and peer IDs, so what does that mean? Let's explain it here - peers can be seen on multiple ip addresses over time, ip addresses can be reused on different peers.

Maybe also mention that we collect protocols and user agents and how we do it.